src/app/pages/delivery-reports/delivery-reports.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { HttpClient, HttpEventType } from '@angular/common/http';
+import { map } from  'rxjs/operators';
+
+@Component({
+    selector: 'ngx-project',
+    templateUrl: './delivery-reports.component.html',
+    styleUrls: ['./delivery-reports.component.scss']
+})
+export class DeliveryReportsComponent {
+
+    projectId: String;
+    form: FormGroup;
+    deliveryReport: Object = null;
+
+    constructor(private route: ActivatedRoute,
+                private formBuilder: FormBuilder,
+                private httpClient: HttpClient) {
+    }
+
+    ngOnInit() {
+        this.form = this.formBuilder.group({
+            excelReportFile: [''],
+            projectId: '',
+        });
+        this.projectId = this.route.snapshot.paramMap.get('projectId');
+        console.log('ngOnInit, projectId', this.projectId);
+    }
+
+    onFileChange(event) {
+        if (event.target.files.length > 0) {
+            const file = event.target.files[0];
+            this.form.get('excelReportFile').setValue(file);
+        }
+    }
+
+    onSubmit() {
+        const formData = new FormData();
+        console.log('Excel', this.form.get('excelReportFile').value)
+        formData.append('deliveryReportFile', this.form.get('excelReportFile').value);
+        this.upload(this.projectId, formData).subscribe(
+          (res) => {
+            this.deliveryReport = res;
+            console.log('Delivery report', this.deliveryReport);
+          },
+          (err) => console.log(err)
+        );
+    }
+
+    public upload(projectId, data) {
+        let uploadURL = `http://localhost:9090/projects/${projectId}/deliveryReports/upload`;
```

Comment:
We should create an environment file and compose the URL based on that. Not urgent but will save some time later Ex:
local: 'localhost:9090',
dev:'domain:9090'
etc

Reply:
Moved URL to enviroment file. Also, I moved the excel upload in ProjectService. If all is good, I'll move the other project related API calls there as well.